src/theme.rs

```diff
@@ -0,0 +1,77 @@
+// insert into the event loop, a watcher for the theme & theme mode for changes
+
+// update a Arc<Mutex<Theme>> in the state on change of the theme and mark all interfaces for a redraw.
+
+use calloop::LoopHandle;
+use cosmic::cosmic_theme::{
+    palette::{self, Srgba},
+    Theme, ThemeMode,
+};
+
+use crate::state::State;
+
+pub(crate) fn clear_color(theme: &Theme<Srgba>) -> [f32; 4] {
+    let neutral_2 = theme.palette.neutral_2;
+    [
+        neutral_2.red,
+        neutral_2.green,
+        neutral_2.blue,
+        neutral_2.alpha,
+    ]
+}
+
+pub(crate) fn group_color(theme: &Theme<Srgba>) -> [f32; 3] {
+    let neutral_8 = theme.palette.neutral_8;
+    [neutral_8.red, neutral_8.green, neutral_8.blue]
+}
+
+pub(crate) fn active_window_hint(theme: &Theme<Srgba>) -> palette::Srgba {
+    if let Some(hint) = theme.window_hint {
+        palette::Srgba::from(hint)
+    } else {
+        theme.accent_color()
+    }
+}
+
+pub fn watch_theme(handle: LoopHandle<'_, State>) -> Result<(), cosmic_config::Error> {
```

Comment:
Hm. We could use `ConfigWatchSource` from libcosmic here, but I guess that would mean 3 channels and calloop sources instead of 1...

I guess we end up spawning a thread here for each `watch` call? Probably the overhead doesn't matter that much.

So that should be fine for now, though it would be nice to figure out how to use epoll+inotify and kqueue to monitor for changes without threads and channels.